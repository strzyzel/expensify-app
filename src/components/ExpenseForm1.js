import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';
import 'react-dates/initialize';
export default class ExpenseForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            description: props.expense ? props.expense.description :  '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount/100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: undefined
        }

    }
    
    
    onTextInputChange = (e) => {
        const state = {}; state[e.target.name] = e.target.value;
        this.setState(() => state);
    };
    
    onAmountChange = (e) => {
        const amount = e.target.value;
        if(!!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ( { amount } ));
        };
    };

    onDateChange = (createdAt) => {
        if(createdAt) this.setState(() => ({ createdAt }));
    };

    onFocusChange = ({focused}) => {
        this.setState(() => ({ calendarFocused: focused }))
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.amount) {
            this.setState(()=> ({ error: "Provide required values." }));
        } else {
            this.setState(()=> ({ error: undefined }));
            this.props.onSubmit({
                'description': this.state.description,
                'amount': parseFloat(this.state.amount, 10)*100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            })
        }
    };
    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    {this.state.error && <p>{this.state.error}</p>}
                    <input
                        type="text"
                        name="description"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onTextInputChange}
                    />
                    <input
                        type="text"
                        name="amount"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker 
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => {
                            return false;
                        }}
                    />
                    <textarea 
                        name="note"
                        placeholder="Add a note for your expense (optional)"
                        value={this.state.note}
                        onChange={this.onTextInputChange}
                    ></textarea>
                    <button>Submit</button>
                </form>
            </div>
        )
    }
}