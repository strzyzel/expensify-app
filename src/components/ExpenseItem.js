import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    return (<div>
        <Link to={`/edit/${id}`}><p>Description: {description}</p></Link>
        <p>Amount: {amount}</p>
        <p>Created at: {new Date(createdAt).toISOString()}</p>
    </div>)
};

export default ExpenseListItem;