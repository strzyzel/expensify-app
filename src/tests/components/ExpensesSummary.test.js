import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should correctly render summary with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={252}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should correctly render summary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={500}/>);
    expect(wrapper).toMatchSnapshot();
});