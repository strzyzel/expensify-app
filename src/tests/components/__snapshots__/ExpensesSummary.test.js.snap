// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render summary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling 
      <span>
        $5.00
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render summary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $2.52
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
